src/PushNotifications.Ports/TopicsSubscriptionsPort.cs

```diff
@@ -109,15 +113,48 @@ public async Task HandleAsync(UnsubscribedFromTopic @event)
                 return;
             }
 
-            bool isSuccessful = true;
             foreach (SubscriptionToken token in projectionReponse.Data.State.Tokens)
             {
                 ITopicSubscriptionManager subscriptionManager = deliveries[token.SubscriptionType];
-                isSuccessful &= await subscriptionManager.UnsubscribeFromTopicAsync(token, @event.Id.Topic);
+                var result = await subscriptionManager.UnsubscribeFromTopicAsync(token, @event.Id.Topic);
+
+                if (result.IsSuccess == false)
+                {
+                    var deviceSubscriptionId = DeviceSubscriptionId.New(@event.Id.NID, token.Token);
+                    RemoveInvalidUnSubscribingToken(result.InvalidTokens, deviceSubscriptionId, @event.Id.SubscriberId, token, @event.Id.Topic);
+                }
+            }
+        }
+
+        private void RemoveInvalidSubscribingToken(bool invalidTokens, DeviceSubscriptionId id, DeviceSubscriberId subscriberId, SubscriptionToken subscriptionToken, Topic topic)
```

Comment:
if you use DeviceSubscriptionId only to get the tenant, you can pass just the tenant as an input parameter of this method instead of the whole object.